packages/@aws-cdk/pipelines/README.md

```diff
@@ -876,6 +876,72 @@ const pipeline = new CdkPipeline(this, 'MyPipeline', {
 });
 ```
 
+### Action Execution Denied
+
+While attempting to deploy an application stage, the "Prepare" or "Deploy" stage may fail with a cryptic error like:
+
+`Action execution failed
+Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: 0123456ABCDEFGH;
+S3 Extended Request ID: 3hWcrVkhFGxfiMb/rTJO0Bk7Qn95x5ll4gyHiFsX6Pmk/NT+uX9+Z1moEcfkL7H3cjH7sWZfeD0=; Proxy: null)`
+
+This generally indicates that the roles necessary to deploy have been deleted (or deleted and re-created);
+for example, if the bootstrap stack has been deleted and re-created, this scenario will happen. Under the hood,
+the resources that rely on these roles (e.g., `cdk-$qualifier-deploy-role-$account-$region`) point to different
+canonical IDs than the recreated versions of these roles, which causes the errors. There are no simple solutions
+to this issue, and for that reason we **strongly recommend** that bootstrap stacks not be deleted and re-created
+once created.
+
+The most automated way to solve the issue is to introduce a secondary bootstrap stack. By changing the qualifier
```

Comment:
Wow, this is a lot of information and detail. Wondering if this is still appropriate here.

As an alternative strategy, how about sticking the detailed instructions in a markdown file in GitHub, and linking to the GItHub location? 

Conditional approval, leaving to you to decide.

Reply:
Would you suggest moving just this new section, or perhaps the entire Troubleshooting section? I think it would make more sense to move the latter, but I worry about lowering visibility and the customer experience of being navigated from the (potentially language-specific) docs to GitHub, especially for our users who don't typically interact with us here.

My vote is to leave it here for now, but have a re-think about the Pipelines docs in general, especially as the new API drops and we have double the content as we currently do.